docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,17 @@
+
+## Technical Debt
+Please be sure to consider how heavily your contribution impacts the maintainability of the project long term, sometimes less is more.  We don't want to merge collossal pull requests with hundreds of dependencies by a driveby contributor.
+
+## Due Diligence Checklist
+Please make sure that you've considered the following before you submit your Pull Requests as ready for merging.
+* I've run Code linters and [gofmt](https://golang.org/cmd/gofmt/) to make sure that my code is readable.
+* I have read through formerly submitted pull requests and git issues issues to make sure that this contribution is required and isn't a duplicate. Also, so that I can manage to close any git Issues needing closed relating to this feature submission.
+* To the best of my ability, I've followed the [CONTRIBUTING.md](./CONTRIBUTING.md) to make sure that I'm up to date on how to best contribute to Stash projects.
```

Comment:
The contributor is already reading the CONTRIBUTING.md is this needed?

Reply:
Likely not I just copied over the points and edited them a little.